Make getVerticesPerPoint method public and improve tests

Blast545 opened the pull request:
Should close #840 

I didn't have to extend existing API because it seems there's already API available to get the number of vertices being used after the render mode is set.

Opening as draft until I check with CI.

Signed-off-by: Jorge Perez <jjperez@ekumenlabs.com>

Blast545 committed:
Make tests less dependant to opengl version

Signed-off-by: Jorge Perez <jjperez@ekumenlabs.com>

Blast545 commented:
Reproducing CI error on linux:
[![Build Status](https://ci.ros2.org/job/ci_linux/16331/badge/icon)](https://ci.ros2.org/job/ci_linux/16331/)

Testing with this PR:
[![Build Status](https://ci.ros2.org/job/ci_linux/16332/badge/icon)](https://ci.ros2.org/job/ci_linux/16332/)

Blast545 committed:
Make method public

Signed-off-by: Jorge Perez <jjperez@ekumenlabs.com>

Blast545 committed:
Fix unused variables

Signed-off-by: Jorge Perez <jjperez@ekumenlabs.com>

Blast545 commented:
Ugh, didn't check the method is private. I had to make it public with e9961f48bf80e91aa5cfa84e3ab0e75d94c688f0, I'll rename the PR after is ready to be merged.

Testing again with this PR:
[![Build Status](https://ci.ros2.org/job/ci_linux/16337/badge/icon)](https://ci.ros2.org/job/ci_linux/16337/)

Blast545 commented:
PTAL @jacobperron

clalancette commented on rviz_rendering/test/rviz_rendering/objects/point_cloud_test.cpp:
```diff
@@ -259,11 +258,7 @@ TEST_F(
-    } else if (glsl_version >= 120) {
-      number_of_vertices_per_box = 6 * 3 * 2;  // six sides with two triangles each
-    }
+    number_of_vertices_per_box = point_cloud->getVerticesPerPoint();
```
Nit: you can combine this statement with the one above, and just do:

```
size_t number_of_vertices_per_box = point_cloud->getVerticesPerPoint();
```

Also, in the loop in line 251 to 254, I think you should also call `getVerticesPerPoint`, as that seems to be better than hard-coding 1.

Blast545 replied:
Addressed with 37d7bf1eeb3adeb0f67f9095021a057fc731380a

clalancette commented on rviz_rendering/test/rviz_rendering/objects/point_cloud_test.cpp:
```diff
@@ -281,15 +276,10 @@ TEST_F(PointCloudTestFixture, addPoints_adds_new_renderable_whenever_it_is_calle
-  } else if (glsl_version >= 120) {
-    number_of_vertices_per_flat_square = 3 * 2;  // two triangles for one square
-  }
+  number_of_vertices_per_flat_square = point_cloud->getVerticesPerPoint();
```
Same comment here about combining this statement with the one above.

Blast545 replied:
Addressed with 5b71a093f6beaf74f5151bdd2974995be6c8cc33

clalancette reviewed (changes requested):
Overall, this looks like an improvement.  I left two small things to fix; once those are done, I'm happy to approve.

Blast545 committed:
Address review comments

Signed-off-by: Jorge Perez <jjperez@ekumenlabs.com>

Blast545 committed:
Address review comments #2

Signed-off-by: Jorge Perez <jjperez@ekumenlabs.com>

Blast545 commented:
Thanks for the review @clalancette !

Full CI after addressing comments:
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=16342)](http://ci.ros2.org/job/ci_linux/16342/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=10954)](http://ci.ros2.org/job/ci_linux-aarch64/10954/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=16706)](http://ci.ros2.org/job/ci_windows/16706/)

clalancette reviewed (approved):
Thanks for the changes, this look good to me.

jacobperron reviewed (approved):
Thanks for looking into this :bow:

clalancette commented:
CI is green, so I'm going to go ahead and merge this.
